### DISCUSSION TITLE
I want to understand what the meta_data field is used for

### DISCUSSION THREAD

**Discussion by sombhattacharyya1983** [ellykits/neat-form#111] [Q&A] [answered]
I was trying to understand the JSOn schema and what the fields are necessary for and I have two queries,

I am not sure I understand what the meta_data field is trying to do ? is that a object in memory at the time of submission of the form. It will be really helpful if you could point me to a simpler example.

Also , I am thinking of extracting a model out of this JSON so it can be represented in the database so I can design a UI for users to create/manage the forms in a WYSIWIG way. Have you given this idea a thought as yet or is there some documentation on how to save this model to a database table ?

**Comment by ellykits** [comment, marked as answer]
Hi @sombhattacharyya1983  The **meta_data** field is optional. It is used when you want to hold extra information for the field that cannot be represented as **field value**. _For example say you have a dropdown spinner showing a list of countries and you want to return the country code, name of the country, population, capital city e.t.c with the form results_, you can use the field meta_data to hold this piece of information. When the form is submitted, they will accompany the field value. You can see how this is used on the field named **country** in the `sample_one_form.json` file in the sample app.

The idea behind neat-form was simply to create a library for generating forms using JSON schema, the rest of the workflow around forms like saving the form data we left it to the users of the library. Neat form however provides you the final data in JSON format, how it saved is up to the users. Besides the JSON format, the developer can work with the form data directly through the `ViewModel` programmatically (Data in a map with field name as the key).

On the client side, this is the model used to represent the JSON,  [NForm](https://github.com/ellykits/neat-form/blob/master/neat-form-core/src/main/java/com/nerdstone/neatformcore/domain/model/NForm.kt). I would recommend using [`SQLDelight`](https://github.com/cashapp/sqldelight), [`Room`](https://developer.android.com/topic/libraries/architecture/room?gclid=CjwKCAiAlNf-BRB_EiwA2osbxSx8PICojPRjGx8n4Yr-fY1pweOd6UQLDGeebjX-EESqelzN4ZHw4xoCCxIQAvD_BwE&gclsrc=aw.ds) or any of your favorite ORMs to save the form data (again this is and will not be part of the library). With regards to creating UI For managing the forms, yes I have had thoughts about it, however there is not much going on as you can see from here [Neat Form Web form builder](https://github.com/ellykits/neat-form-builder). Working on the form builder was kinda deprioritized for now, for lack of time to commit to it plus we do not have much of it's use case at the moment. It will be great to share your ideas, and your contributions are welcome.
